src/key/factory.js

```diff
@@ -61,9 +62,11 @@ export async function generate(options) {
   options = helper.sanitizeKeyOptions(options);
   options.subkeys = options.subkeys.map(function(subkey, index) { return helper.sanitizeKeyOptions(options.subkeys[index], options); });
 
-  let promises = [helper.generateSecretKey(options)];
-  promises = promises.concat(options.subkeys.map(helper.generateSecretSubkey));
-  return Promise.all(promises).then(packets => wrapKeyObject(packets[0], packets.slice(1), options));
```

Comment:
There are a few places where Promises are mixed with async code making code less readable than a style that is either just promises or just async. I changed this one but left a number of these unchaged